doc/reference/kernel/other/interrupts.rst

```diff
@@ -152,6 +152,28 @@ The IRQ must be subsequently **enabled** to permit the ISR to execute.
     Disabling an IRQ prevents *all* threads in the system from being preempted
     by the associated ISR, not just the thread that disabled the IRQ.
 
+Zero Latency Interrupts
+-----------------------
+
+    Preventing interruptions by applying an IRQ lock may increase the observed
+    interrupt latency. A high interrupt latency, however, may not be acceptable
+    for certain low-latency use-cases.
+
+    The kernel addresses such use-cases by allowing interrupts with critical
+    latency constraints to execute at a priority level that cannot be blocked
+    by interrupt locking. These interrupts are defined as
+    *zero-latency interrupts*. The feature is enabled using
+    :option:`CONFIG_ZERO_LATENCY_IRQS`.
```

Comment:
When people are reading this they might say "OK, so that's how I enable it; what do I do next?".  A link to `ISR_DECLARE_DIRECT()` would answer that question; here,  in the description of the option, or both.

Reply:
Hm, right, I will rephrase this. I only wanted to say that the KConfig option enables the support of the feature.

Reply:
updated